### ISSUE TITLE
`scw run docker` doesn't give a public IP

### ISSUE THREAD

**Issue by ndarilek** [scaleway/scaleway-cli#347]
According to `scw run -h`:

```
  --ip-address=""       Assign a reserved public IP, a 'dynamic' one or 'none' (default to 'none' if gateway specified, 'dynamic' otherwise)
```

I installed the binary using the `go get` method about 6 hours ago, and `scw run docker` does not assign a public IP. eth0 comes up with something in the 10.XXX network. I do see LXC failures launching the LXC network bridge, but this is literally my first experience with Scaleway so I don't know if that's an issue with the Docker image or if this tool is acting up. Am I misunderstanding how the `--ip-address` flag works?

As an aside, I asked support this question and was told:
"we don't provide support on softwares part unfortunately"
Does that mean your support doesn't provide support for the software under this Github organization? New to Scaleway and trying to decide whether to build anything on it, but my first experience has been a couple broken boots and some errors which I don't know whether they're in Scaleway or scw. Then support says you don't support software. :)

Thanks.

**Comment by moul** [comment]
Hi @ndarilek, thank you for this question, it will help us to improve our documentation,

> eth0 comes up with something in the 10.XXX network

with or without a public ip address, eth0 will always be a 10.X.X.X ip address, your public ip address will be nated

you can easily check if you have a public ip address by trying to curl/ping something on internet

See https://www.scaleway.com/faq/server/network/ for more info

> Does that mean your support doesn't provide support for the software under this Github organization? 

I think they understood you had a specific problem with the docker image, not with scaleway-cli

For images, we are giving best effort support, especially on GitHub (public, and helpful for other people), for scaleway-cli, we will fix every bugs

> but my first experience has been a couple broken boots and some errors which I don't know whether they're in Scaleway or scw

I'm very interested if you have more info :)

**Comment by ndarilek** [comment]
Got it, so this is just how the service works. Is it at all possible to 
discover the IP from the machine's own interfaces itself? If for 
instance I'm running a service that updates a Cloudflare DNS record, 
it'd need to discover its local IP via the networking stack. It wouldn't 
ping out or use an external service, the service isn't written to do that.

Thanks for this explanation. Support just said they didn't support the 
software and left it at that, they didn't explain how eth0 works 
specific to Scaleway. Are there any docs that explain this and other 
differences? I've run services on traditional VPSs for nearly a decade 
so have plenty of experience with those, but Scaleway obviously differs 
in how it handles IPs, the control plain needing to be hit via HTTP at 
points, etc.

**Comment by moul** [comment]
> Is it at all possible to discover the IP from the machine's own interfaces itself?

I’ve no idea about doing it from the itself, maybe there are some IP aliasing options I don’t know yet.

From BASH you can easily get your public IP address by running the following command: `scw-metadata PUBLIC_IP_ADDRESS`.

Information also available at `http://169.254.42.42/conf` and `http://169.254.42.42/conf?format=json`.

> If for instance I'm running a service that updates a Cloudflare DNS record

To avoid losing your IP address, you can use a reserved one, see more here: https://www.scaleway.com/docs/deal-with-private-reserved-ips/.

I personally use Cloudflare DNS for some personal projects: I create CNAME entries pointing to the `<UUID>.pub.cloud.scaleway.com` url provided on the web console, this url won't change even if the public IP address does.

> Support just said they didn't support the software and left it at that, they didn't explain how eth0 works
specific to Scaleway. Are there any docs that explain this and other differences?

Sorry for the misunderstanding, they thought you were talking about a specific problem with Docker

For the docs, take a look at https://www.scaleway.com/docs/?q=network

> I've run services on traditional VPSs for nearly a decade so have plenty of experience with those, but Scaleway obviously differs in how it handles IPs, the control plain needing to be hit via HTTP at points, etc.

Yes we have some differences with other providers, we fount multiple workarounds to mimics the virtualization features on dedicated hardware.

We recently launched our own VPS offer with virtualization, we kept some of our workarounds (for instance, the HTTP calls during boot) to keep service similar across all our offers, so you can migrate from a VPS offer to a dedicated server while keeping your image with your personal data

**Comment by QuentinBrosse** [comment]
This issue has been automatically closed because it has not had recent activity. Feel free to reopen it if you think that it is necessary.
Note that we are working on a totally new version of the CLI to support all the new coming products.
Thank you for your contributions and stay tuned! :wink:
